src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNamesTests.cs

```diff
@@ -0,0 +1,1127 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class ConstructorParametersShouldMatchPropertyAndFieldNamesTests
+    {
+        #region Class Props Do Not Match Referenced Parameter Names
+
+        [Fact]
+        public async Task CA1071_ClassPropsDoNotMatch_ConstructorParametersShouldMatchPropertyNames_CSharp()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    public int FirstProp { get; }
+                    public object SecondProp { get; }
+
+                    [JsonConstructor]
+                    public C1(int {|#0:firstDrop|}, object {|#1:secondDrop|})
+                    {
+                        this.FirstProp = firstDrop;
+                        this.SecondProp = secondDrop;
+                    }
+                }",
+                CA1071CSharpPropertyOrFieldResultAt(0, "C1", "firstDrop", "FirstProp"),
+                CA1071CSharpPropertyOrFieldResultAt(1, "C1", "secondDrop", "SecondProp"));
```

Comment:
As long as the source has diagnostics, codefix should be tested, even if it's not offered.